policy/modules/system/systemd.te

```diff
@@ -1132,3 +1137,39 @@ files_var_filetrans(systemd_update_done_t, systemd_update_run_t, file, ".updated
 seutil_read_file_contexts(systemd_update_done_t)
 
 systemd_log_parse_environment(systemd_update_done_t)
+
+#########################################
+#
+# User session (systemd --user) local policy
+#
+
+allow systemd_user_session_type self:capability { dac_override sys_resource };
+allow systemd_user_session_type self:process setfscreate;
+allow systemd_user_session_type self:udp_socket create_socket_perms;
+allow systemd_user_session_type self:unix_stream_socket create_stream_socket_perms;
+allow systemd_user_session_type self:netlink_kobject_uevent_socket { bind create getattr setopt };
+
+allow systemd_user_session_type systemd_user_runtime_t:sock_file { create write };
+userdom_user_runtime_filetrans(systemd_user_session_type, systemd_user_runtime_t, sock_file)
+
+dev_write_sysfs_dirs(systemd_user_session_type)
+dev_read_sysfs(systemd_user_session_type)
+
+files_read_etc_files(systemd_user_session_type)
+files_list_usr(systemd_user_session_type)
+
+fs_getattr_cgroup(systemd_user_session_type)
+fs_rw_cgroup_files(systemd_user_session_type)
+fs_manage_cgroup_dirs(systemd_user_session_type)
+
+kernel_read_kernel_sysctls(systemd_user_session_type)
+
+mount_list_runtime(systemd_user_session_type)
+
+storage_getattr_fixed_disk_dev(systemd_user_session_type)
+
+# for systemd to read udev status
+udev_read_pid_files(systemd_user_session_type)
+udev_list_pids(systemd_user_session_type)
+
+userdom_manage_user_runtime_dirs(systemd_user_session_type)
```

Comment:
I know Stephen said that selinux is not optimized for name-based type transitions. but they provide real benefits. particularly for unconfined_t. With these you can to a large extend ensure that the labels of objects of unconfined users are consistent with the labels of  confined users. So that if you ever remap a user from unconfined to user_t for example, that in theory, you do not have to relabel the users home. But that means that you have to use name-based type transitions consistently where possible and that you add the type transition rules to unconfined as well. In dssp2 i have a huge list of these:

https://github.com/DefenSec/dssp2-standard/blob/master/policy/system/u/unconfined.cil

Expensive, yes, but it makes the labels so much more consistent

Reply:
If you do not do the above then a `restorecon -RvF /home` will never be completely silent when you have a system with confined and unconfined users alike